core/api/daemon/src/mill/api/daemon/MillURLClassLoader.scala

```diff
@@ -6,17 +6,25 @@ import java.net.URLClassLoader
  * Convenience wrapper around `java.net.URLClassLoader`. Allows configuration
  * of shared class prefixes, and tracks creation/closing to help detect leaks
  */
-class MillURLClassLoader(
+class MillURLClassLoader private (
```

Comment:
This is probably what's causing MIMA to fail. We can't privatize the constructor of a class that was public before